src/devices/machine/cs8900a.cpp

```diff
@@ -0,0 +1,1522 @@
+                        cs8900_ia_mac[3],cs8900_ia_mac[4],cs8900_ia_mac[5]);
+#endif
+        break;
+    }
+}
+
+/*
+ This is called *before* the relevant octets are read
+*/
+void cs8900a_device::cs8900_sideeffects_read_pp(u16 ppaddress,int odd_address)
+{
+    switch (ppaddress)
+    {
+    case CS8900_PP_ADDR_SE_RXEVENT:
+        /* reading this before all octets of the frame are read
+           performs an "implied skip" */
+        {
+            int access_mask = (odd_address) ? 1 : 2;
+            
+            /* update the status register only if the full word of the last
+               status was read! unfortunately different access patterns are
+               possible: either the status is read LH, LH, LH...
+               or HL, HL, HL, or even L, L, L or H, H, H */
+            if((access_mask & rxevent_read_mask)!=0) {
+                /* receiver is not enabled */
+                if(!rx_enabled) {
+#ifdef CS8900_DEBUG_WARN_RXTX
+                    log_message(cs8900_log,"WARNING! Can't receive any frame (Receiver is not enabled)!");
+#endif
+                } else {
+                    /* perform frame reception */
+                    u16 ret_val = cs8900_receive();
+
+                    /* RXSTATUS and RXEVENT are the same, except that RXSTATUS buffers
+                       the old value while RXEVENT sets a new value whenever it is called
+                    */
+                    SET_PP_16(CS8900_PP_ADDR_RXSTATUS,   ret_val); 
+                    SET_PP_16(CS8900_PP_ADDR_SE_RXEVENT, ret_val); 
+                }
+                
+                /* reset read mask of (possible) other access */
+                rxevent_read_mask = access_mask;
+            } else {
+                /* add access bit to mask */
+                rxevent_read_mask |= access_mask;
+            }
+        }
+
+        break;
+
+    case CS8900_PP_ADDR_SE_BUSST:
+        if(odd_address) {
+            /* read busst before transmit condition is fullfilled */
+            if(tx_state == CS8900_TX_GOT_LEN) {
+                u16 bus_status = GET_PP_16(CS8900_PP_ADDR_SE_BUSST);
+                /* check Rdy4TXNow flag */
+                if((bus_status & 0x100) == 0x100) {
+                    tx_state = CS8900_TX_READ_BUSST;
+#ifdef CS8900_DEBUG_RXTX_STATE
+                    log_message(cs8900_log, "TX: Ready4TXNow set! (%04x)",
+                                bus_status);
+#endif
+                } else {
+#ifdef CS8900_DEBUG_RXTX_STATE
+                    log_message(cs8900_log, "TX: waiting for Ready4TXNow! (%04x)",
+                                bus_status);
+#endif                    
+                }
+            }
+        }
+        break;
+    }
+}
+
+/* ------------------------------------------------------------------------- */
+/* read/write from packet page register */
+
+/* read a register from packet page */
+u16 cs8900a_device::cs8900_read_register(u16 ppaddress)
```

Comment:
This function is using a different indentation style to the rest of the file.  Please use one tab per scope level.